gutils/gioftp.c

```diff
@@ -690,14 +690,14 @@ void *GIO_dispatch(GIOControl *gc) {
     (stdfuncs->PostSuccess)(gc);
     if ( ctl!=-1 ) close(ctl);
     free(path);
-    free(host); free(username); free(password);
+    free(host); free(username); free(password); free(had);
```

Comment:
These don't look safe to me. If you look at lines 618-622, `had` might be `calloc`ed or it might be pulled from `addr`. To free it you'll have to track which. 

*However*, reading the code it doesn't appear that `had` is even used anymore. It doesn't appear after line 625. So you could just rip out lines 617-627 (no need to grab the lock when not examining anything), and the other `had` references.

Reply:
Thank you for your feedback!
Indeed, I had seen the calloc but overlooked the "addr->had".
You're also right, "had" isn't used, just initialized, so let's remove it completely, it'll simplify code and remove a potential leak.

I'll do this when I get back home after my daytime job.

Reply:
Just noticed I could edit directly from github! :-)
Follow jtanx's comment, I've just removed all the changes from gioftp.c since he's gonna completely remove the file.